domains/certificates/Query.API/API/DataSyncSyncer/DataSyncService.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+using System.Security.Claims;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using API.DataSyncSyncer.Dto;
+using CertificateEvents.Primitives;
+using Marten;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Tokens;
+
+namespace API.DataSyncSyncer;
+
+public class DataSyncService
+{
+    private readonly HttpClient httpClient;
+    private readonly ILogger<DataSyncService> logger;
+    private Dictionary<string, DateTimeOffset>? periodStartTimeDictionary;
+
+    public DataSyncService(HttpClient httpClient, ILogger<DataSyncService> logger)
+    {
+        this.httpClient = httpClient;
+        this.logger = logger;
+    }
+
+    public void SetState(Dictionary<string, DateTimeOffset> state)
+    {
+        periodStartTimeDictionary = state;
+    }
+
+    public async Task<List<DataSyncDto>> FetchMeasurements(string GSRN, string meteringPointOwner,
+        CancellationToken cancellationToken)
+    {
+        var dateFrom = periodStartTimeDictionary[GSRN].ToUnixTimeSeconds();
+
+        var now = DateTimeOffset.UtcNow;
+        var midnight = new DateTimeOffset(now.Year, now.Month, now.Day, 0, 0, 0, TimeSpan.Zero).ToUnixTimeSeconds();
+        var result = new List<DataSyncDto>();
+
+        if (dateFrom < midnight)
+        {
+            try
+            {
+                result = await RequestDatasync(
+                    GSRN,
+                    new Period(
+                        DateFrom: dateFrom,
+                        DateTo: midnight
+                    ),
+                    meteringPointOwner,
+                    cancellationToken
+                );
+            }
+            catch (Exception e)
+            {
+                logger.LogInformation("An error occured: {error}, no measurements was fetched", e.Message);
+            }
+
+        }
+
+        SetNextPeriodStartTime(result, GSRN);
+        return result;
+    }
+
+    private void SetNextPeriodStartTime(List<DataSyncDto> measurements, string GSRN)
+    {
+        if (measurements.IsEmpty())
+        {
+            return;
+        }
+
+        var newestMeasurement = measurements.Max(m => m.DateTo);
+        periodStartTimeDictionary![GSRN] = DateTimeOffset.FromUnixTimeSeconds(newestMeasurement);
+    }
+
+    private async Task<List<DataSyncDto>> RequestDatasync(string gsrn, Period period, string meteringPointOwner,
+        CancellationToken cancellationToken)
+    {
+        logger.LogInformation(
+            "Fetching data in period from {from} to: {to}", period.DateFrom.ToString("MM/dd/yy H:mm:ss"), period.DateTo.ToString("MM/dd/yy H:mm:ss")
```

Comment:
`Perdod.DateFrom` and `Period.DateTo` are of type "long" and this will not print a date.

This: `DateTimeOffset.UtcNow.ToUnixTimeSeconds().ToString("MM/dd/yy H:mm:ss")` will return the string "MM/dd/yy H:mm:ss"

If you want to print the timestamp as something to easily read, then you have to parse it back to DateTimeOffset and then call `.ToString("o")` on that.